sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala

```diff
@@ -222,3 +222,11 @@ private[sql] object AnyTimestampType extends AbstractDataType with Serializable
 
   def unapply(e: Expression): Boolean = acceptsType(e.dataType)
 }
+
+/**
+ * The interval type which conforms to the ANSI SQL standard.
+ *
+ * @since 3.2.0
+ */
+@Unstable
```

Comment:
do we need this annotation? it's `private[sql]` and not a public API

Reply:
+1

Reply:
ok. Let me remove `@Unstable`.